pallets/api/src/fungibles/mod.rs

```diff
@@ -218,24 +277,31 @@ pub mod pallet {
 			spender: AccountIdOf<T>,
 			value: BalanceOf<T>,
 		) -> DispatchResultWithPostInfo {
-			let who = ensure_signed(origin.clone())
+			let owner = ensure_signed(origin.clone())
 				.map_err(|e| e.with_weight(Self::weight_approve(0, 0)))?;
-			let current_allowance = AssetsOf::<T>::allowance(id.clone(), &who, &spender);
-			let spender = T::Lookup::unlookup(spender);
-			let id: AssetIdParameterOf<T> = id.into();
+			let current_allowance = AssetsOf::<T>::allowance(id.clone(), &owner, &spender);
+			let spender_source = T::Lookup::unlookup(spender.clone());
+			let id_param: AssetIdParameterOf<T> = id.clone().into();
 
 			if value.is_zero() {
```

Comment:
Why not have this check directly after the origin check? There are unnecessary reads which occur which dont factor into this check, so more efficient to return as early as possible.

Reply:
Yes good one

[refactor: fungible pallet](https://github.com/r0gue-io/pop-node/pull/153/commits/3b3ae9fcf4de6115bc5a2ecb9a92a2ca2e0d0068)